### ISSUE TITLE
Visidata on macOS fails from ftpfs.py test

### ISSUE THREAD

**Issue by bensleveritt** [NixOS/nixpkgs#100595]
**Describe the bug**
Attempting to install Visidata on macOS fails, specifically due to a failing python3.8-fs test. I'm relatively new to Nix, so I'm not sure how to isolate the actual failing package (I'm not sure what python38-fs is. Is it a lib for python, but installed via Nix? It's not listed in the [expression](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/misc/visidata/default.nix))

**To Reproduce**
1. On macOS: `nix-env -iA nixpkgs.visidata`
2. Note failure

**Expected behavior**
Visidata installs correctly

**Additional context**

<detail>
    <summary>Installation log</summary>

```sh
installing 'visidata-1.5.2'
these derivations will be built:
  /nix/store/vkd45qqc12hrqcignbk4gc25z2g7d6sn-python3.8-fs-2.4.11.drv
  /nix/store/mf697ilciffdg3zwmppfdgy5kckjh84j-python3.8-fonttools-4.14.0.drv
  /nix/store/cavy12xpivckcs1g7dxnysycs8s3347x-visidata-1.5.2.drv
building '/nix/store/vkd45qqc12hrqcignbk4gc25z2g7d6sn-python3.8-fs-2.4.11.drv'...
Sourcing python-recompile-bytecode-hook.sh
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing setuptools-check-hook
unpacking sources
unpacking source archive /nix/store/gjzvnc1j8pzbbqwms0rgz0y728d375qh-fs-2.4.11.tar.gz
source root is fs-2.4.11
setting SOURCE_DATE_EPOCH to timestamp 1567868228 of file fs-2.4.11/setup.cfg
patching sources
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/fs
copying fs/memoryfs.py -> build/lib/fs
copying fs/mountfs.py -> build/lib/fs
copying fs/time.py -> build/lib/fs
copying fs/tree.py -> build/lib/fs
copying fs/iotools.py -> build/lib/fs
copying fs/enums.py -> build/lib/fs
copying fs/mirror.py -> build/lib/fs
copying fs/_typing.py -> build/lib/fs
copying fs/_version.py -> build/lib/fs
copying fs/subfs.py -> build/lib/fs
copying fs/_url_tools.py -> build/lib/fs
copying fs/_fscompat.py -> build/lib/fs
copying fs/tools.py -> build/lib/fs
copying fs/constants.py -> build/lib/fs
copying fs/filesize.py -> build/lib/fs
copying fs/appfs.py -> build/lib/fs
copying fs/_ftp_parse.py -> build/lib/fs
copying fs/_repr.py -> build/lib/fs
copying fs/osfs.py -> build/lib/fs
copying fs/__init__.py -> build/lib/fs
copying fs/copy.py -> build/lib/fs
copying fs/test.py -> build/lib/fs
copying fs/mode.py -> build/lib/fs
copying fs/glob.py -> build/lib/fs
copying fs/wildcard.py -> build/lib/fs
copying fs/walk.py -> build/lib/fs
copying fs/lrucache.py -> build/lib/fs
copying fs/ftpfs.py -> build/lib/fs
copying fs/_bulk.py -> build/lib/fs
copying fs/compress.py -> build/lib/fs
copying fs/move.py -> build/lib/fs
copying fs/permissions.py -> build/lib/fs
copying fs/wrapfs.py -> build/lib/fs
copying fs/errors.py -> build/lib/fs
copying fs/tempfs.py -> build/lib/fs
copying fs/info.py -> build/lib/fs
copying fs/zipfs.py -> build/lib/fs
copying fs/wrap.py -> build/lib/fs
copying fs/path.py -> build/lib/fs
copying fs/tarfs.py -> build/lib/fs
copying fs/base.py -> build/lib/fs
copying fs/error_tools.py -> build/lib/fs
copying fs/multifs.py -> build/lib/fs
creating build/lib/fs/opener
copying fs/opener/memoryfs.py -> build/lib/fs/opener
copying fs/opener/registry.py -> build/lib/fs/opener
copying fs/opener/appfs.py -> build/lib/fs/opener
copying fs/opener/osfs.py -> build/lib/fs/opener
copying fs/opener/__init__.py -> build/lib/fs/opener
copying fs/opener/ftpfs.py -> build/lib/fs/opener
copying fs/opener/errors.py -> build/lib/fs/opener
copying fs/opener/tempfs.py -> build/lib/fs/opener
copying fs/opener/parse.py -> build/lib/fs/opener
copying fs/opener/zipfs.py -> build/lib/fs/opener
copying fs/opener/tarfs.py -> build/lib/fs/opener
copying fs/opener/base.py -> build/lib/fs/opener
copying fs/py.typed -> build/lib/fs
installing to build/bdist.macosx-10.6-x86_64/wheel
running install
running install_lib
creating build/bdist.macosx-10.6-x86_64
creating build/bdist.macosx-10.6-x86_64/wheel
creating build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/memoryfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/mountfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/time.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/tree.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/iotools.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/enums.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/mirror.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/_typing.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/_version.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/subfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/_url_tools.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/_fscompat.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/tools.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/constants.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/filesize.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/appfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/_ftp_parse.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/_repr.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/osfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/__init__.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/copy.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/test.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/mode.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/glob.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
creating build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/memoryfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/registry.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/appfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/osfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/__init__.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/ftpfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/errors.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/tempfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/parse.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/zipfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/tarfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/opener/base.py -> build/bdist.macosx-10.6-x86_64/wheel/fs/opener
copying build/lib/fs/wildcard.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/walk.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/lrucache.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/ftpfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/_bulk.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/compress.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/move.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/permissions.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/py.typed -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/wrapfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/errors.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/tempfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/info.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/zipfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/wrap.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/path.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/tarfs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/base.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/error_tools.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
copying build/lib/fs/multifs.py -> build/bdist.macosx-10.6-x86_64/wheel/fs
running install_egg_info
running egg_info
writing fs.egg-info/PKG-INFO
writing dependency_links to fs.egg-info/dependency_links.txt
writing requirements to fs.egg-info/requires.txt
writing top-level names to fs.egg-info/top_level.txt
reading manifest file 'fs.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'fs.egg-info/SOURCES.txt'
Copying fs.egg-info to build/bdist.macosx-10.6-x86_64/wheel/fs-2.4.11-py3.8.egg-info
running install_scripts
creating build/bdist.macosx-10.6-x86_64/wheel/fs-2.4.11.dist-info/WHEEL
creating 'dist/fs-2.4.11-py2.py3-none-any.whl' and adding 'build/bdist.macosx-10.6-x86_64/wheel' to it
adding 'fs/__init__.py'
adding 'fs/_bulk.py'
adding 'fs/_fscompat.py'
adding 'fs/_ftp_parse.py'
adding 'fs/_repr.py'
adding 'fs/_typing.py'
adding 'fs/_url_tools.py'
adding 'fs/_version.py'
adding 'fs/appfs.py'
adding 'fs/base.py'
adding 'fs/compress.py'
adding 'fs/constants.py'
adding 'fs/copy.py'
adding 'fs/enums.py'
adding 'fs/error_tools.py'
adding 'fs/errors.py'
adding 'fs/filesize.py'
adding 'fs/ftpfs.py'
adding 'fs/glob.py'
adding 'fs/info.py'
adding 'fs/iotools.py'
adding 'fs/lrucache.py'
adding 'fs/memoryfs.py'
adding 'fs/mirror.py'
adding 'fs/mode.py'
adding 'fs/mountfs.py'
adding 'fs/move.py'
adding 'fs/multifs.py'
adding 'fs/osfs.py'
adding 'fs/path.py'
adding 'fs/permissions.py'
adding 'fs/py.typed'
adding 'fs/subfs.py'
adding 'fs/tarfs.py'
adding 'fs/tempfs.py'
adding 'fs/test.py'
adding 'fs/time.py'
adding 'fs/tools.py'
adding 'fs/tree.py'
adding 'fs/walk.py'
adding 'fs/wildcard.py'
adding 'fs/wrap.py'
adding 'fs/wrapfs.py'
adding 'fs/zipfs.py'
adding 'fs/opener/__init__.py'
adding 'fs/opener/appfs.py'
adding 'fs/opener/base.py'
adding 'fs/opener/errors.py'
adding 'fs/opener/ftpfs.py'
adding 'fs/opener/memoryfs.py'
adding 'fs/opener/osfs.py'
adding 'fs/opener/parse.py'
adding 'fs/opener/registry.py'
adding 'fs/opener/tarfs.py'
adding 'fs/opener/tempfs.py'
adding 'fs/opener/zipfs.py'
adding 'fs-2.4.11.dist-info/LICENSE'
adding 'fs-2.4.11.dist-info/METADATA'
adding 'fs-2.4.11.dist-info/WHEEL'
adding 'fs-2.4.11.dist-info/top_level.txt'
adding 'fs-2.4.11.dist-info/RECORD'
removing build/bdist.macosx-10.6-x86_64/wheel
Finished executing setuptoolsBuildPhase
installing
Executing pipInstallPhase
/private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/dist /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11
Processing ./fs-2.4.11-py2.py3-none-any.whl
Requirement already satisfied: appdirs~=1.4.3 in /nix/store/ykn7m69g59dmjjq9afxhkdp5xggb0l2j-python3.8-appdirs-1.4.4/lib/python3.8/site-packages (from fs==2.4.11) (1.4.4)
Requirement already satisfied: pytz in /nix/store/55kprl0vfj92rnjbm4fv3q14mqyiix9f-python3.8-pytz-2020.1/lib/python3.8/site-packages (from fs==2.4.11) (2020.1)
Requirement already satisfied: six~=1.10 in /nix/store/4kif7c2kh08aj4bfwc8a17pnlfzc7b5z-python3.8-six-1.15.0/lib/python3.8/site-packages (from fs==2.4.11) (1.15.0)
Requirement already satisfied: setuptools in /nix/store/zj94d8na5qz3fvjk8zqrhjcrvdvpmsqr-python3.8-setuptools-47.3.1/lib/python3.8/site-packages (from fs==2.4.11) (47.3.1.post20200907)
Installing collected packages: fs
Successfully installed fs-2.4.11
/private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11
Finished executing pipInstallPhase
post-installation fixup
strip is /nix/store/ip78z0sjkhafnqq6zckm3zx4n3lcf3q3-cctools-binutils-darwin-927.0.2/bin/strip
stripping (with command strip and flags -S) in /nix/store/3a5xa98i4vh436rxf4dv7znshh2bk4sk-python3.8-fs-2.4.11/lib 
patching script interpreter paths in /nix/store/3a5xa98i4vh436rxf4dv7znshh2bk4sk-python3.8-fs-2.4.11
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
============================= test session starts ==============================
platform darwin -- Python 3.8.5, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11, inifile: setup.cfg
collected 1687 items / 2 deselected / 1685 selected                            

tests/test_base.py .                                                     [  0%]
tests/test_copy.py ...............                                       [  0%]
tests/test_errors.py ....                                                [  1%]
tests/test_ftp_parse.py .....                                            [  1%]
tests/test_ftpfs.py ..........s......................................... [  4%]
.....................................E....s............................. [  8%]
......................................................s................. [ 13%]
.........................................................                [ 16%]
tests/test_info.py .........                                             [ 16%]
tests/test_iotools.py ............                                       [ 17%]
tests/test_lrucache.py .                                                 [ 17%]
tests/test_memoryfs.py ................................................. [ 20%]
..............................                                           [ 22%]
tests/test_mode.py ..                                                    [ 22%]
tests/test_mountfs.py .................................................. [ 25%]
........................................................................ [ 29%]
.........................................                                [ 32%]
tests/test_move.py ..                                                    [ 32%]
tests/test_multifs.py .................................................. [ 35%]
........................s............                                    [ 37%]
tests/test_osfs.py ....s................................................ [ 40%]
.................................                                        [ 42%]
tests/test_path.py .....................                                 [ 43%]
tests/test_permissions.py ..............                                 [ 44%]
tests/test_subfs.py ....s............................................... [ 47%]
......................................s................................. [ 52%]
.................................................                        [ 54%]
tests/test_tarfs.py .....s.............................................. [ 58%]
...............................s........................................ [ 62%]
.....................................s.................................. [ 66%]
...........................................s............................ [ 70%]
.................................................s...................... [ 75%]
........................................................................ [ 79%]
...............                                                          [ 80%]
tests/test_tempfs.py ....s.............................................. [ 83%]
.......................................s................................ [ 87%]
...................................................                      [ 90%]
tests/test_time.py ..                                                    [ 90%]
tests/test_tools.py ...                                                  [ 90%]
tests/test_tree.py ....                                                  [ 91%]
tests/test_walk.py ....................                                  [ 92%]
tests/test_wildcard.py ...                                               [ 92%]
tests/test_wrap.py ..                                                    [ 92%]
tests/test_wrapfs.py ...                                                 [ 92%]
tests/test_zipfs.py .....s.............................................. [ 95%]
.....................................................................    [100%]

==================================== ERRORS ====================================
________________ ERROR at teardown of TestFTPFS.test_writetext _________________

cls = <class 'tests.test_ftpfs.TestFTPFS'>

    @classmethod
    def tearDownClass(cls):
>       cls.server.stop()

tests/test_ftpfs.py:166: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/wb44np6335sq0bmjlcbnjp3395hshhig-python3.8-pyftpdlib-1.5.6/lib/python3.8/site-packages/pyftpdlib/test/__init__.py:355: in stop
    assert_free_resources()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def assert_free_resources():
        ts = threading.enumerate()
        assert len(ts) == 1, ts
        p = psutil.Process()
        children = p.children()
        if children:
            for p in children:
                p.kill()
                p.wait(1)
>           assert not children, children
E           AssertionError: [psutil.Process(pid=47258, status='terminated', exitcode=<Negsignal.SIGKILL: -9>, started='11:20:56')]

/nix/store/wb44np6335sq0bmjlcbnjp3395hshhig-python3.8-pyftpdlib-1.5.6/lib/python3.8/site-packages/pyftpdlib/test/__init__.py:261: AssertionError
=============================== warnings summary ===============================
tests/test_base.py:11
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_base.py:11: PytestCollectionWarning: cannot collect test class 'TestFS' because it has a __init__ constructor (from: tests/test_base.py)
    class TestFS(FS):

tests/test_ftpfs.py: 3 tests with warnings
tests/test_memoryfs.py: 1 test with warning
tests/test_mountfs.py: 2 tests with warnings
tests/test_multifs.py: 1 test with warning
tests/test_osfs.py: 1 test with warning
tests/test_subfs.py: 2 tests with warnings
tests/test_tarfs.py: 5 tests with warnings
tests/test_tempfs.py: 2 tests with warnings
tests/test_zipfs.py: 1 test with warning
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/fs/test.py:1064: DeprecationWarning: Please use assertRaisesRegex instead.
    with self.assertRaisesRegexp(errors.ResourceNotFound, error_msg):

tests/test_ftpfs.py::TestAnonFTPFS::test_appendbytes
  /nix/store/wb44np6335sq0bmjlcbnjp3395hshhig-python3.8-pyftpdlib-1.5.6/lib/python3.8/site-packages/pyftpdlib/authorizers.py:243: RuntimeWarning: write permissions assigned to anonymous user.
    warnings.warn("write permissions assigned to anonymous user.",

tests/test_iotools.py::TestIOTools::test_make_stream
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_iotools.py:29: DeprecationWarning: Please use assertTrue instead.
    self.assert_(isinstance(data, bytes))

tests/test_path.py::TestPathFunctions::test_isdotfile
tests/test_path.py::TestPathFunctions::test_isdotfile
tests/test_path.py::TestPathFunctions::test_isdotfile
tests/test_path.py::TestPathFunctions::test_isdotfile
tests/test_path.py::TestPathFunctions::test_isdotfile
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:181: DeprecationWarning: Please use assertTrue instead.
    self.assert_(isdotfile(path))

tests/test_path.py::TestPathFunctions::test_iswildcard
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:204: DeprecationWarning: Please use assertTrue instead.
    self.assert_(iswildcard("*"))

tests/test_path.py::TestPathFunctions::test_iswildcard
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:205: DeprecationWarning: Please use assertTrue instead.
    self.assert_(iswildcard("*.jpg"))

tests/test_path.py::TestPathFunctions::test_iswildcard
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:206: DeprecationWarning: Please use assertTrue instead.
    self.assert_(iswildcard("foo/*"))

tests/test_path.py::TestPathFunctions::test_iswildcard
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:207: DeprecationWarning: Please use assertTrue instead.
    self.assert_(iswildcard("foo/{}"))

tests/test_path.py::TestPathFunctions::test_recursepath
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:155: DeprecationWarning: Please use assertEqual instead.
    self.assertEquals(recursepath("/"), ["/"])

tests/test_path.py::TestPathFunctions::test_recursepath
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:156: DeprecationWarning: Please use assertEqual instead.
    self.assertEquals(recursepath("hello"), ["/", "/hello"])

tests/test_path.py::TestPathFunctions::test_recursepath
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:157: DeprecationWarning: Please use assertEqual instead.
    self.assertEquals(recursepath("/hello/world/"), ["/", "/hello", "/hello/world"])

tests/test_path.py::TestPathFunctions::test_recursepath
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:158: DeprecationWarning: Please use assertEqual instead.
    self.assertEquals(

tests/test_path.py::TestPathFunctions::test_recursepath
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:161: DeprecationWarning: Please use assertEqual instead.
    self.assertEquals(recursepath("hello", reverse=True), ["/hello", "/"])

tests/test_path.py::TestPathFunctions::test_recursepath
  /private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-python3.8-fs-2.4.11.drv-0/fs-2.4.11/tests/test_path.py:162: DeprecationWarning: Please use assertEqual instead.
    self.assertEquals(recursepath("", reverse=True), ["/"])

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== short test summary info ============================
ERROR tests/test_ftpfs.py::TestFTPFS::test_writetext - AssertionError: [psuti...
==== 1670 passed, 15 skipped, 2 deselected, 36 warnings, 1 error in 45.53s =====
/nix/store/np0b7p04kbhl244yv65d2sc37kk7wcqf-python3-3.8.5/lib/python3.8/multiprocessing/resource_tracker.py:96: UserWarning: resource_tracker: process died unexpectedly, relaunching.  Some resources might leak.
  warnings.warn('resource_tracker: process died unexpectedly, '
Traceback (most recent call last):
  File "/nix/store/np0b7p04kbhl244yv65d2sc37kk7wcqf-python3-3.8.5/lib/python3.8/multiprocessing/resource_tracker.py", line 201, in main
    cache[rtype].remove(name)
KeyError: '/mp-vsd0gsdt'
Traceback (most recent call last):
  File "/nix/store/np0b7p04kbhl244yv65d2sc37kk7wcqf-python3-3.8.5/lib/python3.8/multiprocessing/resource_tracker.py", line 201, in main
    cache[rtype].remove(name)
KeyError: '/mp-40w_n_jo'
Traceback (most recent call last):
  File "/nix/store/np0b7p04kbhl244yv65d2sc37kk7wcqf-python3-3.8.5/lib/python3.8/multiprocessing/resource_tracker.py", line 201, in main
    cache[rtype].remove(name)
KeyError: '/mp-ewqz51j2'
Traceback (most recent call last):
  File "/nix/store/np0b7p04kbhl244yv65d2sc37kk7wcqf-python3-3.8.5/lib/python3.8/multiprocessing/resource_tracker.py", line 201, in main
    cache[rtype].remove(name)
KeyError: '/mp-ieuqap3k'
Traceback (most recent call last):
  File "/nix/store/np0b7p04kbhl244yv65d2sc37kk7wcqf-python3-3.8.5/lib/python3.8/multiprocessing/resource_tracker.py", line 201, in main
    cache[rtype].remove(name)
KeyError: '/mp-9_nvy71a'
Traceback (most recent call last):
  File "/nix/store/np0b7p04kbhl244yv65d2sc37kk7wcqf-python3-3.8.5/lib/python3.8/multiprocessing/resource_tracker.py", line 201, in main
    cache[rtype].remove(name)
KeyError: '/mp-8o_ct1p7'
builder for '/nix/store/vkd45qqc12hrqcignbk4gc25z2g7d6sn-python3.8-fs-2.4.11.drv' failed with exit code 1
cannot build derivation '/nix/store/mf697ilciffdg3zwmppfdgy5kckjh84j-python3.8-fonttools-4.14.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/cavy12xpivckcs1g7dxnysycs8s3347x-visidata-1.5.2.drv': 1 dependencies couldn't be built
error: build of '/nix/store/cavy12xpivckcs1g7dxnysycs8s3347x-visidata-1.5.2.drv' failed
```
</detail>

**Notify maintainers**
@7c6f434c 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.5.0, macOS 10.15.5`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.7`
 - channels(horatio): `"home-manager, darwin, nixpkgs-21.03pre243353.6d4b93323e7"`
 - nixpkgs: `/Users/horatio/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by 7c6f434c** [comment]
`fs` is an optional dependency of `fonttools`, an optional dependency of `visidata`

I guess you could make `fs` dependency of `fonttools` conditional on the platform not being `x86_64-darwin`.

**Comment by bensleveritt** [comment]
Thanks for the info. Still blundering around somewhat, but [should something like this work](https://github.com/bensleveritt/nixpkgs/commit/2412b4ff2eae8c080adce6464400da8fbc79b6c7)?

I've not been able to test it as `nix-build -A python38Packages.fonttools` from the root nixpkg checkout fails:

```
...
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:460: tificc] Error 1
make[1]: Leaving directory '/private/var/folders/ld/jx0rtyjx17b49vq5tftv6_wc0000gn/T/nix-build-lcms2-2.11.drv-0/lcms2-2.11/utils/tificc'
test_state_abbr (tests.providers.test_address.TestEnUS) ... ok
make: *** [Makefile:476: all-recursive] Error 1
builder for '/nix/store/3wvqd8846fm70ryzr2c1wyz3b4i9x2in-lcms2-2.11.drv' failed with exit code 2
```

Is there something special about `nix-build`, Python and macOS or is there something wrong with my Nix?

**Comment by 7c6f434c** [comment]
I would put the optionals in parentheses and you need a `++` afterwards. Not sure how the linked version ends up getting evaluated, might be something strange

**Comment by raboof** [comment]
> I've not been able to test it as `nix-build -A python38Packages.fonttools` from the root nixpkg checkout fails:

Is that on master? You might want to checkout `nixpkgs-unstable` - the problem you quote seems to be the main reason `nixpkgs-unstable` hasn't been bumped up to a more recent version yet.

**Comment by bensleveritt** [comment]
Ah ha, thanks @raboof, I was on Master. And on switching to unstable, it seems to have been resolved, so I updated my installation of Nixpkgs (to `nixpkgs-21.03pre246624.cfed29bfcb2`) and this issue has been resolved since `nixpkgs-21.03pre243353.6d4b93323e7`.

Thanks all!

**Comment by raboof** [comment]
:+1: good to hear! tracking the issue on master at #101068 - if you feel up to it it might be great if you could help diagnose/fix it :laughing:
